Python/optimizer_analysis.c

```diff
@@ -412,6 +416,9 @@ optimize_uops(
 error:
     DPRINTF(3, "\n");
     DPRINTF(1, "Encountered error in abstract interpreter\n");
+    if (opcode <= MAX_UOP_ID) {
```

Comment:
This is somewhat concerning, on one hand it's bounded by 512, then here it's bounded by MAX_UOP_ID. MAX_UOP_ID might grow to above 512, so I think bounding by either one and sticking to it is better.